testing.md

```diff
@@ -0,0 +1,71 @@
+# Testing
+
+## Why write tests?
+
+We write automated tests to allow us to quickly demonstrate that our software meeting business requirements. They're one of a few tools we use to increase our confidence in our work prior to deployment.
+
+## When do I write tests?
+
+### Test necessity
+_"If this test fails, what business requirement is compromised?"_
+
+This is a good question to ask yourself before writing tests. If the answer is _"None"_, you shouldn't be writing it. If the answer is _"I don't know"_, it's possible the test you're writing has no value.
+
+### Net costs
+It's also worth considering the net cost of tests before you write them. For example, if the business impact of a regression is small and unlikely enough, it could be that you're probably incurring a net cost to the business by writing a test that needs to be **1.** understood, **2.** run as part of your CI and **3.** maintained.
```

Comment:
Historically I've seen a lot of tests very tightly coupled to the code they're testing. As in, if I were to change the code in a mediumly significant way, I'd have to spend a LOT of time trying to piece together all of the failing tests it introduces. If we write our tests in the right way, the cost of maintenance drops towards the point at which it becomes negligible. In the ideal world, test maintenance is a non-issue.